File: D2L1/SRMCards.java
import java.util.Arrays;

public class SRMCards {
    public int maxTurns(int[] cards) {
        Arrays.sort(cards); 
        int ret = 0, prev = 0;
        for (int i = 0; i < cards.length; ++i) {
            if (cards[i] != prev) {
                ++ret;
                prev = cards[i] + 1;
            }
        }
        return ret;
    }
}

File: D2L1/SRMCards.cpp
#include <vector>
using namespace std;

int maxTurns(vector<int> const &cards) {
    bool a[502]; 
    fill(a, a + 502, false);
    
    for (int i = 0; i < cards.size(); ++i)
        a[cards[i]] = true;

    int ret = 0;
    for (int i = 1; i <= 500; ++i) 
        if (a[i]) {
            if (a[i + 1]) ++i;
            ++ret;
        }        

    return ret;
}
